server/ugRestApi.js

```diff
@@ -0,0 +1,154 @@
+const { server: serverConfig } = require('./configuration')
+
+const groupNames = (courseCode, semester, ladokRoundIds) => ({
+  // Used to get examiners and responsibles from UG Rest Api
+  teachers: ladokRoundIds.map(
+    round => `edu.courses.${String(courseCode).slice(0, 2)}.${courseCode}.${semester}.${round}.teachers`
+  ),
+  examiners: [`edu.courses.${String(courseCode).slice(0, 2)}.${courseCode}.examiner`],
+  responsibles: ladokRoundIds.map(
+    round => `edu.courses.${String(courseCode).slice(0, 2)}.${courseCode}.${semester}.${round}.courseresponsible`
+  ),
+  assistants: ladokRoundIds.map(
+    round => `edu.courses.${String(courseCode).slice(0, 2)}.${courseCode}.${semester}.${round}.assistants`
+  ), // edu.courses.SF.SF1624.20191.1.assistants
+})
+
+const getCurrentDateTime = () => {
+  const today = new Date()
+  const date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate()
+  const time = today.getHours() + ':' + today.getMinutes() + ':' + today.getSeconds()
+  const dateTime = date + ' ' + time
+  return dateTime
+}
+
+const createPersonHtml = personList => {
+  let personString = ''
+  personList &&
+    personList.forEach(person => {
+      if (person) {
+        personString += `<p class = "person">
+      <img src="https://www.kth.se/files/thumbnail/${person.username}" alt="Profile picture" width="31" height="31">
+      <a href="/profile/${person.username}/" property="teach:teacher">
+          ${person.givenName} ${person.lastName ? person.lastName : person.surname} 
+      </a> 
+    </p>  `
+      }
+    })
+  return personString
+}
+
+/**
+ * This will first prepare filter query then pass that query to ug rest api and then return groups data along with members.
+ * @param assistants Assistants group name
+ * @param teachers Teachers group name
+ * @param examiners Examiners group name
+ * @param responsibles Responsibles group name
+ * @returns Will return groups along with members from ug rest api.
+ */
+async function getAllGroupsAlongWithMembersRelatedToCourse(assistants, teachers, examiners, responsibles) {
+  const { url, key } = serverConfig.ugRestApiURL
+  const { authTokenURL, authClientId, authClientSecret } = serverConfig.ugAuth
+  const ugConnectionProperties = {
+    ugTokenURL: authTokenURL,
+    clientId: authClientId,
+    clientSecret: authClientSecret,
+    ugURL: url,
+    subscriptionKey: key,
+  }
+  ugRestApiHelper.initConnectionProperties(ugConnectionProperties)
+  const filterData = []
+  if (assistants.length) {
+    filterData.push(assistants[0])
+  }
+  if (teachers.length) {
+    filterData.push(teachers[0])
+  }
+  if (examiners.length) {
+    filterData.push(examiners[0])
+  }
+  if (responsibles.length) {
+    filterData.push(responsibles[0])
+  }
+  log.info('Started fetching data of groups along with members from UG Rest Api at ', getCurrentDateTime())
+  log.info('Fetching groups along with members')
+  log.info('Using Filter Query Data', { filterData })
```

Comment:
What is included in filterData? if it is user data, we don't want to push to graycloud log data, because it's personal data, like email (GDPR rules). it is better to print the names of groups and then how many users were fetched,

Reply:
In old code f.e., printed how many users fetched for courseCode and semester     
```
log.info(
      ' Ug Redis fetched correctly,  number of teachers: ',
      arrWithStringifiedArrays[0].length,  {courseCode, semester}
    )
```

Reply:
Added these logs